infra/gravity/config.go

```diff
@@ -15,6 +19,52 @@ import (
 	"gopkg.in/go-playground/validator.v9"
 )
 
+// OS represents OS vendor/version
+type OS struct {
+	Vendor, Version string
+}
+
+// Split returns os vendor and version
+func (os *OS) UnmarshalJSON(b []byte) error {
+	str, err := strconv.Unquote(string(b))
+	if err != nil {
+		return trace.ConvertSystemError(err)
+	}
+	split := strings.Split(str, ":")
+	if len(split) != 2 {
+		return trace.BadParameter("OS should be in format vendor:version, got %q", b)
+	}
+	os.Vendor = split[0]
+	os.Version = split[1]
+	return nil
+}
+
+// Key returns back to serialized form
+func (os *OS) String() string {
+	return fmt.Sprintf("%s:%s", os.Vendor, os.Version)
+}
+
+type StorageDriver string
+
+func (drv *StorageDriver) UnmarshalJSON(b []byte) error {
+	name, err := strconv.Unquote(string(b))
+	if err != nil {
+		return trace.ConvertSystemError(err)
+	}
+	switch name {
+	case constants.DeviceMapper, constants.Overlay, constants.Overlay2, constants.Loopback, constants.ManifestStorageDriver:
```

Comment:
We don't have `loopback` as a constant and it cannot be specified as a value for `--storage-driver` atm - it is implicitly implied if `devicemapper` is used w/o a device